.internal/copyArrayWithOwnProperty.js

```diff
@@ -7,7 +7,7 @@
  * @returns {Array} Returns `array`.
  */
 function copyArrayWithOwnProperty(source, array) {
-  const length = source.length
+  const length = source?.length ?? 0
   array || (array = new Array(length))
```

Comment:
We should check wether both the params are arrays or not first as discussed in the comments as well.

Reply:
Yes, i fixed it